compat/src/index.js

```diff
@@ -80,18 +80,56 @@ class ContextProvider {
  * @param {object | null | undefined} props
  */
 function Portal(props) {
-	let wrap = h(ContextProvider, { context: this.context }, props.vnode);
+	let _this = this;
 	let container = props.container;
+	let wrap = h(ContextProvider, { context: _this.context }, props.vnode);
+
+	// When we change container we should clear our old container and
+	// indicate a new mount.
+	if (_this.container && _this.container !== container) {
+		if (_this.temp.parentNode) _this.container.removeChild(_this.temp);
+		_unmount(_this.wrap);
+		_this.hasMounted = false;
+	}
 
-	if (props.container !== this.container) {
-		hydrate('', container);
-		this.container = container;
+	// When props.vnode is undefined/false/null we are dealing with some kind of
+	// conditional vnode. This should not trigger a render.
+	if (props.vnode) {
+		if (!_this.hasMounted) {
+			// Create a placeholder that we can use to insert into.
+			_this.temp = document.createTextNode('');
+			// Hydrate existing nodes to keep the dom intact, when rendering
+			// wrap into the container.
+			hydrate('', container);
+			// Insert before first child (will just append if firstChild is null).
+			container.insertBefore(_this.temp, container.firstChild);
+			// At this point we have mounted and should set our container.
+			_this.hasMounted = true;
+			_this.container = container;
+			// Render our wrapping element into temp.
+			preactRender(wrap, container, _this.temp);
+		}
+		else {
+			// When we have mounted and the vnode is present it means the
+			// props have changed or a parent is triggering a rerender.
+			// This implies we only need to call render.
+			render(wrap, container);
```

Comment:
Why render here instead of preactRender?  If there isn't a difference, this conditional could be reduced:

```js
let temp;
if (props.vnode) {
  if (!_this.hasMounted) {
    temp = document.createTextNode('');
    hydrate('', container);
    container.insertBefore(temp, container.firstChild);
    _this.hasMounted = true;
    _this.container = container;
  }
  preactRender(wrap, container, temp);
}
```

Reply:
Using `replaceNode` here isn't possible since this can only be used on initially rendering